### DISCUSSION TITLE
deny_all is deprecated even though it seems like it's still part of the code base

### DISCUSSION THREAD

**Discussion by khalidibrahimio** [apache/airflow#36448] [General]
Hello everyone

Seems like the `auth_backends` option `deny_all` is deprecated by [this PR](https://github.com/apache/airflow/pull/21640). But it is being preserved in the codebase still as can be seen [here](https://github.com/apache/airflow/blob/main/airflow/api/auth/backend/deny_all.py) and [here](https://github.com/apache/airflow/blob/main/airflow/www/extensions/init_jinja_globals.py#L80) . Are there plans to remove it? Or is it still usable?

Thank you 🙏🏼

**Comment by potiuk** [comment]
No. You misunderstoood deprecation warning. What is deprecated is using **just** `deny_all` without also adding `session` as well. you should specify both `deny_all` and `session` in your configuration. This way all regular API requests will be denied, while all the API calls from the UI will be allowed.

**Reply by khalidibrahimio** [reply]
Oh okayy, I could've never guessed that. So, `session` only allows API call from the UI right? If that's the case, how is it different from having CORS enabled with the URL = the airflow host domain?

**Reply by potiuk** [reply]
It has nothing to do with CORS. It's authentication on the server side. CORS is signalling the client that requests can be originated from one page to another page. Those two have nothing to do in common.

**Reply by khalidibrahimio** [reply]
Oh okay, I'm sorry then. I don't get what the session option adds to the functionality if `deny_all` is enabled. What do you mean by `all the API calls from the UI will be allowed.`
And thank you for helping me clarify this 🙏🏼

**Reply by potiuk** [reply]
It means that if UI make an API call from within browser it will be allowed. Otherwise your UI will not work because it will not be able to retrieve data from teh webserver via API. This is why `session` is currently added always if you do not have it there (and deprecation warning asks you to add it to your configuration).
